src/Console/BaseCommand.php

```diff
@@ -47,6 +47,15 @@ abstract class BaseCommand implements CommandInterface, EventDispatcherInterface
      */
     protected string $name = 'cake unknown';
 
+    /**
+     * Constructor
+     *
+     * @param \Cake\Console\CommandFactoryInterface $factory Command factory instance.
+     */
+    public function __construct(protected ?CommandFactoryInterface $factory = null)
+    {
+    }
```

Comment:
Now that I know, that constructors can be overwritten with different args I can tell, that this is fine and doesn't break all currently available commands which already use DI.

BUT does this mean, that the property in here still get populated when creating the command?

Reply:
No, if the constructor doesn't have the argument for the factory instance and doesn't call the parent constructor (or set the property) then it will remain null.

I have accounted for that in executeCommand().

Reply:
Interesting, does php yell if it's not implicitly set to null?

Reply:
Hmm.. Not sure, I'll change it to the old fashioned property declaration to be safe.